go-controller/pkg/ovn/egressip_test.go

```diff
@@ -248,7 +248,6 @@ var _ = ginkgo.Describe("OVN master EgressIP Operations", func() {
 
 				fakeOvn.fakeExec.AddFakeCmdsNoOutputNoError(
 					[]string{
-						fmt.Sprintf("ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=_uuid find logical_router_policy priority=%s nexthop!=[]", types.EgressIPReroutePriority),
```

Comment:
can you add a commit message detailing how this commit fixes the issue? It looks like fake exec commands are just being moved around, and i dont see how that fixes it.

Reply:
Moreover, could you link to an observed flake? I have been building master regularly  lately and I have not observed any flakes at all for what concerns the egress IP unit tests.

Reply:
@trozet @alexanderConstantinescu  observed flake is here: https://github.com/ovn-org/ovn-kubernetes/issues/2503 

Most of the tests sets up the fake commands and calls these two functions:

- fakeOvn.controller.WatchEgressNodes()
- fakeOvn.controller.WatchEgressIP()

If you look at the code, in WatchEgressNodes(), we do:
- first, oc.initClusterEgressPolicies() which calls
--- **ovn-nbctl --timeout=15 --may-exist lr-policy-add ovn_cluster_router 101 ip4.src == 10.128.0.0/14 && ip4.dst == 10.128.0.0/14 allow**
- next we call the AddFunc() which calls this if the node is assignable and ready
--- **ovn-nbctl --timeout=15 set logical_switch_port etor-GR_node1 options:nat-addresses=router**

If you look at the code, in WatchEgressIP() we do:
- syncEgressIPs() which calls findLegacyReroutePolicyIDs() and that calls
--- **ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=_uuid find logical_router_policy priority=%s nexthop!=[]**

So, going by the above flow of code the order of fake commands should be:

```
ovn-nbctl --timeout=15 --may-exist lr-policy-add ovn_cluster_router 101 ip4.src == 10.128.0.0/14 && ip4.dst == 10.128.0.0/14 allow
ovn-nbctl --timeout=15 set logical_switch_port etor-GR_node1 options:nat-addresses=router
ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=_uuid find logical_router_policy priority=%s nexthop!=[]
```

but instead it is

```
ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=_uuid find logical_router_policy priority=%s nexthop!=[]
ovn-nbctl --timeout=15 --may-exist lr-policy-add ovn_cluster_router 101 ip4.src == 10.128.0.0/14 && ip4.dst == 10.128.0.0/14 allow
ovn-nbctl --timeout=15 set logical_switch_port etor-GR_node1 options:nat-addresses=router
```

Reply:
The order of executed commands do not matter in `egressip_test.go` because of https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/egressip_test.go#L161 

In any case, I fixed a egress IP unit test flake some time ago: https://github.com/ovn-org/ovn-kubernetes/pull/2428 and I can't really inspect the CI job you linked to, since it happened too long ago and the logs have been rotated. If you think there's still a problem, then I don't oppose this. But I don't understand exactly what it fixes.

Reply:
fwiw I just saw this flake on my laptop: 
```
Summarizing 1 Failure:

[Fail] OVN master EgressIP Operations syncEgressIP for IPv4 [It] should update invalid assignments with unhostable IP on a node 
/home/trozet/go/src/github.com/ovn-kubernetes/go-controller/pkg/ovn/ovn_test.go:120

Ran 121 of 121 Specs in 153.486 seconds
FAIL! -- 120 Passed | 1 Failed | 0 Pending | 0 Skipped
```